cmd/release-controller/types.go

```diff
@@ -338,3 +370,23 @@ func (a tagReferencesByAge) Less(i, j int) bool {
 }
 func (a tagReferencesByAge) Len() int      { return len(a) }
 func (a tagReferencesByAge) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+func calculateBackoff(retryCount int, initialTime *metav1.Time) time.Duration {
```

Comment:
You should be using the `wait.Backoff` type to do this calculation.

Or add a unit test here that validates your logic is correct (and never call `time.Now()` inside a function like this, always pass it in so you can unit test it)